aii-core/src/main/perl/Shellfe.pm

```diff
@@ -323,6 +340,41 @@ sub _initialize
     if ($self->option(INCLUDE)) {
         unshift(@INC, split(/:+/, $self->option(INCLUDE)));
     }
+
+    # Initialization and validation of some options whose default values are based on other options
+    if ( $self->option(NBPDIR_PXELINUX) ) {
+        $self->{CONFIG}->set(NBPDIR_PXELINUX, undef) if $self->option(NBPDIR_PXELINUX) eq NBPDIR_VARIANT_DISABLED;
+    }
+    if ( $self->option(NBPDIR_GRUB2) ) {
+        $self->{CONFIG}->set(NBPDIR_GRUB2, undef) if $self->option(NBPDIR_GRUB2) eq NBPDIR_VARIANT_DISABLED;
+    } else {
+        if ( $self->option(NBPDIR_PXELINUX) ) {
+            my $efi_dir = dirname($self->option(NBPDIR_PXELINUX)) . OSINSTALL_DEF_GRUB2_DIR;
+            $self->{CONFIG}->set(NBPDIR_GRUB2, $efi_dir);
+        } else {
+            $self->{CONFIG}->set(NBPDIR_GRUB2, OSINSTALL_DEF_ROOT_PATH . OSINSTALL_DEF_GRUB2_DIR);
+        }
+    }
+    if ( !$self->option(GRUB2_EFI_KERNEL_ROOT) ) {
+        if ( $self->option(NBPDIR_GRUB2) ) {
+            my $kernel_root = dirname($self->option(NBPDIR_GRUB2));
+            $kernel_root =~ s/^\/\w+//;
+            $kernel_root = '' if ( $kernel_root eq '/' );
+            $self->{CONFIG}->set(GRUB2_EFI_KERNEL_ROOT, $kernel_root);
+        }
+    }
+    # GRUB2_EFI_INITRD_CMD is always derived from GRUB2_EFI_LINUX_CMD as
+    # Grub2 has a set of linux/initrd command pairs that must match together.
+    if ( $self->option(GRUB2_EFI_LINUX_CMD) ) {
+        my $initrd_cmd = $self->option(GRUB2_EFI_LINUX_CMD);
+        $initrd_cmd =~ s/linux/initrd/;
+        if ( $initrd_cmd eq GRUB2_EFI_INITRD_CMD ) {
+            $self->error("Commands to load the kernel and initrd are identical ($initrd_cmd)");
+        }
+        $self->{CONFIG}->define(GRUB2_EFI_INITRD_CMD);
+        $self->{CONFIG}->set(GRUB2_EFI_INITRD_CMD, $initrd_cmd);
```

Comment:
if you can configure in the config file, it's better that this is a regular option, with help etc etc